pointgrey_camera_driver/src/stereo_nodelet.cpp

```diff
@@ -271,9 +281,14 @@ class PointGreyStereoCameraNodelet: public nodelet::Nodelet
       {
         sensor_msgs::ImagePtr image(new sensor_msgs::Image);
         sensor_msgs::ImagePtr second_image(new sensor_msgs::Image);
-        pg_.grabStereoImage(*image, frame_id_, *second_image, second_frame_id_);
+        uint32_t hw_timestamp = 0;
+        pg_.grabStereoImage(*image, frame_id_, *second_image, second_frame_id_, &hw_timestamp);
 
         ros::Time time = ros::Time::now();
+        if(device_time_translator_){
+          time = device_time_translator_->update(hw_timestamp, time, 0.0); // The image->header.stamp is always zero for the Bumblebee 2 (Firewire).
```

Comment:
The Bumblebee comment is a little bit random here?

Reply:
I agree. But I wanted to document somehow why I do this hack. I have no idea whether this is different for other stereo cameras. Over time one could maintain a list of devices having his issue and use `image->header.stamp` for all the others. Maybe I should write as much as a comment ..